File: services/auth/src/main/java/com/example/iter/auth/service/model/OAuthExchangeResult.java
package com.example.iter.auth.service.model;

import com.example.iter.auth.dto.response.OAuthActionRequiredResponse;

public sealed interface OAuthExchangeResult {

    record Authenticated(IssuedTokenPair tokenPair) implements OAuthExchangeResult {
    }

    record ActionRequired(OAuthActionRequiredResponse response) implements OAuthExchangeResult {
    }
}

File: services/auth/src/main/kotlin/com/example/iter/auth/service/model/OAuthExchangeResult.kt
package com.example.iter.auth.service.model

import com.example.iter.auth.dto.response.OAuthActionRequiredResponse

sealed interface OAuthExchangeResult {

    @JvmRecord
    data class Authenticated(val tokenPair: IssuedTokenPair) : OAuthExchangeResult

    @JvmRecord
    data class ActionRequired(val response: OAuthActionRequiredResponse) : OAuthExchangeResult
}
